app/controllers/DepartureTestMessagesController.scala

```diff
@@ -108,4 +112,35 @@ class DepartureTestMessagesController @Inject()(cc: ControllerComponents,
               InternalServerError
           }
     }
+
+  def submitDepartureDeclaration: Action[NodeSeq] = (authAction andThen channelAction).async(parse.xml) {
+    implicit request =>
+      departureConnector
+        .createDeclarationMessage(request.body, request.channel)
+        .map {
+          response =>
+            response.status match {
+              case status if is2xx(status) =>
+                response.header(LOCATION) match {
+                  case Some(locationValue) =>
+                    val departureId = Utils.lastFragment(locationValue)
+                    Accepted(
+                      Json.toJson(
+                        HateaosDepartureResponse(
+                          DepartureId(departureId.toInt),
+                          MessageType.DepartureDeclaration,
+                          request.body,
+                          locationValue
+                        )))
```

Comment:
This seems completely out of place considering that the rest of this API is for generating the XML and sending that on, depending on the message type desired and any additional instructions. This is practically a duplicate of the public API with a flexible channel.

In addition, from the look of the associated ticket, you're going to need many more responses including a cancellation, and whilst I can understand not needing the generation, why isn't this simply an endpoint that takes in XML, and maybe a DepartureId (obviously not for the creation) and then forward the XML based on the type? We don't need to have unique endpoints for each message type, but we've a unique one for departures.

Further, in that context, why isn't this being done via the frontend? The ticket mentions Simulate (for a lot of messages),

Reply:
@RobertLaverick As we are using the **test support** in the departure journey test to check the status of submission and successive receiving of NCTS messages we are maintaining the same response type so we can use the same code rather than having lots of conditions to do the same.

Reply:
Simulate means directly sending XML to this new endpoint to create departure declaration rather than the UI flow. As it is tested separately.

Reply:
But you just send back the initial XML - The only useful information here to the test that is new is the 200 and the Id (and arguably the location value).

You've not addressed the concern about the other message type - DepartureCancellation that need to be handled according to your ticket - or for that matter the HATEAOS Response. What value is that giving you for the frontend tests when HATEAOS is only used by the Public API?

Reply:
We are not simulating DepartureCancellation, it is a UI journey  and HATEAOS is because the EIS response is coming in that pattern so keeping the same style for this new end point.

Reply:
Sorry, I completely blanked that HATEAOS wasn't a new introduction.

Fair enough, that makes more sense then.